code/modules/mob/living/basic/guardian/guardian_types/gaseous.dm

```diff
@@ -0,0 +1,153 @@
+	. = ..()
+	if(!isnull(summoner))
+		UnregisterSignal(summoner, COMSIG_ATOM_PRE_PRESSURE_PUSH)
+
+/mob/living/basic/guardian/gaseous/manifest_effects()
+	. = ..()
+	if (!isnull(summoner))
+		RegisterSignal(summoner, COMSIG_ATOM_PRE_PRESSURE_PUSH, PROC_REF(pre_pressure_moved))
+
+/// We stand firm in the face of gas
+/mob/living/basic/guardian/gaseous/proc/pre_pressure_moved(datum/source)
+	SIGNAL_HANDLER
+	return COMSIG_ATOM_BLOCKS_PRESSURE
+
+
+/// Expel a range of gases
+/datum/action/cooldown/mob_cooldown/expel_gas
+	name = "Release Gas"
+	desc = "Start or stop expelling a selected gas into the environment."
+	button_icon = 'icons/mob/actions/actions_spells.dmi'
+	button_icon_state = "smoke"
+	cooldown_time = 0 SECONDS // We're here for the interface not the cooldown
+	melee_cooldown_time = 0 SECONDS
+	click_to_activate = FALSE
+	/// Gas being expelled.
+	var/active_gas = null
+	/// Associative list of types of gases to moles we create every life tick.
+	var/static/list/possible_gases = list(
+		/datum/gas/oxygen = 50,
+		/datum/gas/nitrogen = 750, //overpressurizing is hard!.
+		/datum/gas/water_vapor = 1, //you need incredibly little water vapor for the effects to kick in
+		/datum/gas/nitrous_oxide = 15,
+		/datum/gas/carbon_dioxide = 50,
+		/datum/gas/plasma = 3,
+		/datum/gas/bz = 10,
+	)
+	/// Associative list of types of gases to particle effect colour.
+	var/static/list/gas_colors = list(
+		/datum/gas/oxygen = "#63BFDD", //color of frozen oxygen
+		/datum/gas/nitrogen = "#777777", //grey (grey)
+		/datum/gas/water_vapor = "#96ADCF", //water is slightly blue
+		/datum/gas/nitrous_oxide = "#FEFEFE", //white like the sprite
+		/datum/gas/carbon_dioxide = "#222222", //black like coal
+		/datum/gas/plasma = "#B233CC", //color of the plasma sprite
+		/datum/gas/bz = "#FAFF00", //color of the bz metabolites reagent
+	)
+
+/datum/action/cooldown/mob_cooldown/expel_gas/Grant(mob/granted_to)
+	. = ..()
+	if (isnull(owner))
+		return
+	RegisterSignal(owner, COMSIG_GUARDIAN_RECALLED, PROC_REF(stop_gas))
+
+/datum/action/cooldown/mob_cooldown/expel_gas/Remove(mob/removed_from)
+	UnregisterSignal(owner, list(COMSIG_GUARDIAN_RECALLED, COMSIG_LIVING_LIFE))
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/expel_gas/Activate(atom/target)
+	StartCooldown(360 SECONDS)
+	// Regeneated each time just in case someone fucks with our list
+	var/list/gas_selection = list("None")
+	for(var/datum/gas/gas as anything in possible_gases)
+		gas_selection[initial(gas.name)] = gas
+
+	var/picked_gas = tgui_input_list(owner, "Select a gas to emit.", "Gas Producer", gas_selection)
+		StartCooldown()
+	if(picked_gas == "None")
+		stop_gas()
+		return
+
+	var/gas_type = gas_selection[picked_gas]
+	if(isnull(picked_gas) || isnull(gas_type))
+		return
+
+	to_chat(owner, span_bolddanger("You start releasing [picked_gas]."))
+	owner.investigate_log("set their gas type to [picked_gas].", INVESTIGATE_ATMOS)
+	var/had_gas = active_gas == null
```

Comment:
```suggestion
	var/had_gas = isnull(active_gas)
```

Reply:
Looking at this it's also just inverted from the check I wanted to make